src/EnergyPlus/BranchInputManager.hh

```diff
@@ -346,7 +349,7 @@ namespace BranchInputManager {
                        Optional_string_const CompName = _  // when mustprint (ScanPlantLoop)  use CompName in error message and scan
     );
 
-    void TestBranchIntegrity(bool &ErrFound);
+    void TestBranchIntegrity(bool &ErrFound);              // ErrFound is a return value, true or false
```

Comment:
It seems like there could be a pattern chosen here where if the value of errors_found is used as a pure return value, and intentionally sets it to either true or false, that could just become the actual return value of the function, instead of a reference argument.  Then in cases where it's just set to true, never set to false, then it could be a reference argument.  From the look of the CHECKED_AND_OKED variable, it looks like there may be some 50 functions that are OK'd, and a big chunk of those could be candidates for this conversion.  It would take a little effort and result in more code changes in this PR to do that now, but it would also make the false-positive frequency reduce.

Let me know what you think about it.  I'd be happy to push on that a little bit right now while this PR is open, otherwise it'll just linger and never get done...

Reply:
I agree that in general if it's a pure return value then the signature should be `bool NameOfFunction()` instead of `void NameOfFunction(bool&)`. It would create somewhat of a shit in typical code patterns in the code base, I don't know if that's acceptable or not, I'll let you be the judge of it.